File: C#/Atbash/AtbashCipher.cs
using System;
using System.Text;
using System.Linq;

public static class AtbashCipher
{
    const string CIPHER = "zyxwvutsrqponmlkjihgfedcba";
    const int BLOCK_SIZE = 5;

    static char EncodeLetter(char c)
    {
        if (!char.IsAsciiLetter(char.ToLower(c)))
        {
            return c;
        }

        int index = Convert.ToInt32(char.ToLower(c)) - 97;
        return CIPHER[index];
    }

    static string CleanString(string s)
    {
        return s.Replace(" ", "").Replace(".", "").Replace(",", "");
    }

    public static string Encode(string plainValue)
    {
        string encoded = "";
        int seperatorCounter = 0;

        foreach (char c in CleanString(plainValue))
        {
            encoded += EncodeLetter(c);
            seperatorCounter++;

            if (seperatorCounter == BLOCK_SIZE)
            {
                encoded += " ";
                seperatorCounter = 0;
            }
        }

        // Just in case it was exactly divisible by 5 and had an extra space put on the end
        return encoded.TrimEnd();
    }

    public static string Decode(string encodedValue)
    {
        string decoded = "";

        foreach (char c in CleanString(encodedValue))
        {
            decoded += EncodeLetter(c);
        }

        return decoded;
    }
}

File: JavaScript/atbash-cipher.js
const CIPHER = "zyxwvutsrqponmlkjihgfedcba";
const BLOCK_SIZE = 5;

const encodeLetter = (letter) => {
    if(!letter.match(/[a-zA-Z]/)) {
        return letter;
    }

    let index = letter.toLowerCase().charCodeAt(0) - 97;
    return CIPHER[index];
}
    
const cleanString = (string) => {
    return string.replaceAll(" ", "").replaceAll(".", "").replaceAll(",", "");
}

const encode = (plainValue) => {
    let clean = cleanString(plainValue)
    let encoded = "";
    let seperatorCounter = 0;

    for(let i = 0; i < clean.length; i++)
    {
        encoded += encodeLetter(clean[i]);
        seperatorCounter++;

        if (seperatorCounter == BLOCK_SIZE)
        {
            encoded += " ";
            seperatorCounter = 0;
        }
    }

    // Just in case it was exactly divisible by 5 and had an extra space put on the end
    return encoded.trimEnd();
};

const decode = (encodedValue) => {
    let decoded = "";
    let clean = cleanString(encodedValue)

    for(let i = 0; i < clean.length; i++) {
        decoded += encodeLetter(clean[i]);
    }

    return decoded;    
};


const original = "The quick brown fox jumps over the lazy dog.";
const encoded = encode(original);
const decoded = decode(encoded);

console.log(`original -> ${original}`);
console.log(`encoded -> ${encoded}`);
console.log(`decoded -> ${decoded}`);